### DISCUSSION TITLE
Common mistakes

### DISCUSSION THREAD

**Discussion by RicheyRyan** [markerikson/react-community-tools-practices-cheatsheet#21] [Ideas]
So as I mentioned in my PR, I've got a bunch I could write and contribute here. @markerikson, you said it wasn't really what you conceived of initially. I'm happy to continue contributing the common mistakes if you are happy to merge them, but if there is a topic you think would be better to cover, I'm up for refocusing my attention. 

From re-reading the README, the original Twitter thread seemed more focused on choosing between the different options in the ecosystem. There are apparent gaps like React state, Mobx, how-tos and use cases. I could contribute confidently to all but Mobx as I don't have a ton of experience with it. Also, in the original thread, you stressed the need for curation, which I agree with, which is why I'm looking to bring my contributions in line with an overall curated vision. Anyway, I'm excited to hear what you think and get started in earnest on this.

**Comment by srmagura** [comment]
I think a slightly-opinionated comparison of the different libraries/frameworks/tools in the ecosystem is very sorely needed, so we should focus our efforts on that for now.

That said, documenting common pitfalls is valuable, and your section on duplicate state was very well written. I think "Common Mistakes" should be kept as part of the site, but perhaps moved towards the bottom of the left navigation bar.

For the type of content to include in the Common Mistakes section, I think we need to keep it deliberately focused on mistakes that are truly common, to keep the scope reasonable.

**Reply by RicheyRyan** [reply]
That sounds fair. I have one more common mistake ready to go so I guess I'll PR that and move the Common Mistakes section at the same time. Then I think I'll have a look at either some of the styling stuff or build tools. 

Starting a new project would cover build tools I suppose? The suggested ways to get started with Webpack, Parcel, Vite, CRA etc etc.

**Reply by srmagura** [reply]
That all sounds good. Also, it may be wise to wait for @markerikson to give his thoughts since he is the mastermind behind all this.

For build tools, I think you could use the same section headings that I used on the [styling pages](https://react-community-tools-practices-cheatsheet.netlify.app/styling/plain-css). The final section could be a mini-tutorial on how to configure the tool to build a minimal React single page app. Maybe the tutorial would cover JavaScript, but there would be a TypeScript version of the config (e.g. `webpack.config.js`) at the end?

I am planning to write the pages on CSS Preprocessors and Emotion, but feel free to tackle CSS Modules or styled-components.

**Reply by RicheyRyan** [reply]
Cool thanks for the heads up and I'll indeed wait for Mark

**Reply by markerikson** [reply]
Yeah, I'd generally echo what Sam's said.

The immediate and primary goal of the site is to provide reference descriptions and overviews of the major tools for a given use case, with tradeoffs.

Other material covering specific usage patterns will likely be helpful, but isn't as important for now.

I think the short-term goal should be to flesh out the "State Management", "Styling", "Build Tools / Frameworks", and "Data Fetching" categories, and also try to nail down final site titles / branding / landing page content.  Once we've got those sorted out, we can start seriously promoting the site and pointing people to it as a meaningful resource.
